grpc/src/main/java/com/linecorp/armeria/internal/server/grpc/AbstractServerCall.java

```diff
@@ -213,24 +213,30 @@ public final void close(Throwable exception) {
     public final void close(Throwable exception, boolean cancelled) {
         exception = Exceptions.peel(exception);
         final Metadata metadata = generateMetadataFromThrowable(exception);
-        final Status status = exceptionHandler.apply(ctx, null, exception, metadata);
-        close(new ServerStatusAndMetadata(status, metadata, false, cancelled), exception);
+        final Status status = Status.fromThrowable(exception);
+        close(status, metadata, cancelled, exception);
     }
 
     @Override
     public final void close(Status status, Metadata metadata) {
-        if (status.getCause() == null) {
-            close(new ServerStatusAndMetadata(status, metadata, false));
+        close(status, metadata, false, null);
+    }
+
+    private void close(Status status, Metadata metadata, boolean cancelled,
+                       @Nullable Throwable originalCauseForLogging) {
```

Comment:
The original naming sounds like I should use it **only** for logging.
```suggestion
                       @Nullable Throwable originalCause) {
```